#! /bin/sh /usr/share/dpatch/dpatch-run
## 19-gcc43.dpatch by Thomas Girard <thomas.g.girard@free.fr>
##
## DP: fix failures and warnings with g++ 4.3
## DP: See http://bugzilla.dre.vanderbilt.edu/show_bug.cgi?id=3270
@DPATCH@
--- ACE_wrappers.orig/ace/config-macros.h
+++ ACE_wrappers/ace/config-macros.h
@@ -380,7 +380,7 @@
 // This eliminates the warnings, and no code is generated for the null
 // conditional statement.  @note that may only be true if -O is enabled,
 // such as with GreenHills (ghs) 1.8.8.
-# define ACE_UNUSED_ARG(a) do {/* null */} while (&a == 0)
+# define ACE_UNUSED_ARG(a) do {/* null */} while (&a < 0)
 #elif defined (__DMC__)
   #define ACE_UNUSED_ID(identifier)
   template <class T>
--- ACE_wrappers.orig/apps/JAWS/server/HTTP_Server.cpp
+++ ACE_wrappers/apps/JAWS/server/HTTP_Server.cpp
@@ -143,10 +143,12 @@
   HTTP_Handler_Factory *f = 0;
 
   if (this->strategy_ != (JAWS::JAWS_POOL | JAWS::JAWS_ASYNCH))
+    {
 	  if (this->caching_)
 		  ACE_NEW_RETURN (f, Synch_HTTP_Handler_Factory (), -1);
 	  else
 		  ACE_NEW_RETURN (f, No_Cache_Synch_HTTP_Handler_Factory (), -1);
+    }
 
   //NOTE: At this point f better not be a NULL pointer,
   //so please do not change the ACE_NEW_RETURN macros unless
